repotracker/github_poller.go

```diff
@@ -187,23 +187,41 @@ func (gRepoPoller *GithubRepositoryPoller) GetRevisionsSince(revision string, ma
 			revisionError = errors.Wrapf(err,
 				"unable to find a suggested merge base commit for revision %v, must fix on projects settings page",
 				revision)
+			gRepoPoller.ProjectRef.RepotrackerError = revisionDetails
+			return []model.Revision{}, revisionError
```

Comment:
Oh we actually aren't updating the projectRef in the DB anymore before returning. We used to do that after both the if and else because both cases had a reason to update the projectRef. If this is the only place in the `!foundLatest` block that we need to update the project, then we can just move the upsert here.

Reply:
After we automatically update the last revision we set the RepotrackerError object back to its original state (not existing).  In the case that the RepotrackerError was not null before we run this update, we'd need to upsert the project after we revert the RepotrackerError to empty.  So I think we might need to update the project ref in the DB after both if and else blocks.

Reply:
That's fine, then we still need to upsert in the if case since we don't reach the upsert.

Reply:
The upsert currently happens after the if/else block completes so I think no change is required? Since upsert is never skipped.

Reply:
Upsert is skipped in the case I have linked here: at 190 we're setting RepotrackerError for our local copy, but then at 191 we return which takes us out of the function completely, so we never reach the Upsert at line 220.

Reply:
Understood.  Decided to also go ahead and avoid updating repotracker error when setting the new revision like you suggested, so I've moved the upsert into the if block as well.